packages/query/src/graphql/plugins/GetMetadataPlugin.ts

```diff
@@ -55,11 +55,12 @@ async function fetchFromApi(): Promise<void> {
   }
 }
 
-async function fetchFromTable(pgClient: any, schemaName: string): Promise<MetaData> {
+async function fetchFromTable(pgClient: any, schemaName: string, chainId: string | undefined): Promise<MetaData> {
   const metadata = {} as MetaData;
   const keys = Object.keys(METADATA_TYPES);
 
-  const {rows} = await pgClient.query(`select key, value from "${schemaName}"._metadata WHERE key = ANY ($1)`, [keys]);
+  const tableName = `_metadata${chainId ? `_${chainId.substring(0, 10)}` : ''}`;
```

Comment:
I don't think we can do `.substring(0, 10)`. Theres no constraint on the length of the chain it and some networks have chain ids < 10 (e.g. `juno-1`)